File: java/data/src/main/java/com/youknow/data/repository/MoviesRepositoryImpl.java
package com.youknow.data.repository;

import com.youknow.data.source.MoviesDataSource;
import com.youknow.domain.repository.MoviesRepository;

public class MoviesRepositoryImpl implements MoviesRepository {

    private MoviesDataSource moviesLocalDataSource;
    private MoviesDataSource moviesRemoteDataSource;

    public MoviesRepositoryImpl(MoviesDataSource localDataSource, MoviesDataSource remoteDataSource) {
        moviesLocalDataSource = localDataSource;
        moviesRemoteDataSource = remoteDataSource;
    }

    @Override
    public void getNowPlaying(MoviesLoadedCallback callback) {
        moviesRemoteDataSource.getNowPlaying(callback);
    }

    @Override
    public void getUpcoming(MoviesLoadedCallback callback) {
        moviesRemoteDataSource.getUpcoming(callback);
    }

    @Override
    public void getMovie(String movieId, MovieLoadedCallback callback) {
        moviesRemoteDataSource.getMovie(movieId, callback);
    }
}

File: kotlin-coroutine-aac/app/src/main/java/com/youknow/movie/data/repository/MoviesRepositoryImpl.kt
package com.youknow.movie.data.repository

import com.youknow.movie.data.source.MoviesDataSource
import com.youknow.movie.domain.model.Movie
import com.youknow.movie.domain.model.SimpleMovie
import com.youknow.movie.domain.repository.MoviesRepository

class MoviesRepositoryImpl(
    private val moviesCacheDataSource: MoviesDataSource,
    private val moviesRemoteDataSource: MoviesDataSource
): MoviesRepository {

    override suspend fun getNowPlaying(): List<SimpleMovie> = moviesRemoteDataSource.getNowPlaying()

    override suspend fun getUpcoming(): List<SimpleMovie> = moviesRemoteDataSource.getUpcoming()

    override suspend fun getMovie(id: String): Movie = moviesRemoteDataSource.getMovie(id)

}